cmd/codegen/generator/go/templates/src/_dagger.gen.go/client.go.tmpl

```diff
@@ -177,3 +177,78 @@ func (c errorWrappedClient) MakeRequest(ctx context.Context, req *graphql.Reques
 	}
 	return nil
 }
+{{/* If it's not in dev mode, we wrap the library client with custom options */}}
+{{ else }}
+
+func Connect(ctx context.Context, opts ...dagger.ClientOpt) (*Client, error) {
+	dag, err := dagger.Connect(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Client{
+		query:  dag.QueryBuilder(),
+		client: dag.GraphQLClient(),
+	}
+
+	if err := serveModuleDepencencies(ctx, c); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+{{ end }}
+
+// serveModuleDependencies services all dependencies of the module.
+// Local dependencies are served by the dagger.json.
+// Remote dependencies are generated by the client generator.
+func serveModuleDepencencies(ctx context.Context, client *Client) error {
+	{{ range $i, $dep := Dependencies -}}
+	if err := client.ModuleSource("{{ $dep.Ref }}", ModuleSourceOpts{RefPin: "{{ $dep.Pin }}"}).WithName("{{ $dep.Name }}").AsModule().Serve(ctx); err != nil {
+		return err
+	}
+	{{ end }}
+
+	modSrc := client.ModuleSource(".")
+	configExist, err := modSrc.ConfigExists(ctx)
+	if err != nil {
+		return err
+	}
+
+	if !configExist {
+		return nil
+	}
+
+	dependencies, err := modSrc.Dependencies(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, dep := range dependencies {
```

Comment:
This is a lot better than it was before :tada: But I think it's still a bit weird to have different code for loading local / git.

IMO the `Dependencies` function should include *all* dependencies - not just git ones. When it comes to loading a local one though, we can modify the path to be relative to the found `dagger.json`.

> [!NOTE]
>
> Aside: we currently can't tell if a dep is local/git based on the config. This is a bit weird actually - we might want to have a `kind` in the module dependency config that we use to load the module. Probably not for now though.

That said, it's all in one place now, which is better. As long as:
1. we add some comments explaining *why* this is separate
2. we rename `Dependencies` to be `GitDependencies` or similar, to clarify the naming.